WcaOnRails/app/assets/javascripts/users.js

```diff
@@ -7,6 +7,7 @@ onPage('users#edit, users#update', function() {
     var delegateStatus = this.value;
     var seniorDelegateRequired = {
       "": false,
+      trainee_delegate: true,
```

Comment:
To answer your question in the issue: when editing a user, we check if a senior delegate is needed for the currently selected delegate status.
The delegate_status is an enum and I think rails uses that to generate the "select" dropdown with all the available statuses.
Upon change, that javascript function is called and shows/hide the senior delegate field if it needs to be filled or not.

Reply:
Okay got it. Thanks :-)